### DISCUSSION TITLE
Dependencies version pinning

### DISCUSSION THREAD

**Discussion by tr11** [aminalaee/sqladmin#134] [Ideas]
Maybe we could switch from using a requirements file to allow using [optional dependencies](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies).  For example, I'm using `sqladmin` with `asyncpg` and `sqlalchemy`, so it'd be nice not having to install `psycopg2` or `aiosqlite` in order to reduce the size of our docker images. Additionally, we could define most of the currently required libs as `dev` extras (`coverage`, `black`, `pytest`, etc).

**Comment by aminalaee** [comment]
Hey, I think I'm not getting your point.

When installing `pip install sqladmin`, it only requires sqlalchemy, starlette, jinja2 and a few others [here](https://github.com/aminalaee/sqladmin/blob/main/setup.cfg#L29).

You won't have to install any database driver like asyncpg or dev dependencies defined in requirements.txt so I don't see how that should affect your docker image size.
If you want to contribute to the project, you need to install the dev dependencies i.e. requirements.txt .

Does that make sense?

**Reply by tr11** [reply]
I guess installing from git was pulling the other ones too. I should have checked that! Forget about my comment :-)

**Comment by aminalaee** [comment]
No worries :)
